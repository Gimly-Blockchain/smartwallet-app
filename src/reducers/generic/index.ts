import { AnyAction } from 'redux'
import { pickBy } from 'ramda'

export const APPWRAP_UPDATE_CONFIG = 'APPWRAP_UPDATE_CONFIG'
export const APPWRAP_SHOW_LOADER = 'APPWRAP_SHOW_LOADER'
export const APPWRAP_SET_LOCKED = 'APPWRAP_SET_LOCKED'
export const APPWRAP_REGISTER_CONFIG = 'APPWRAP_REGISTER_CONFIG'
export const APPWRAP_UNREGISTER_CONFIG = 'APPWRAP_UNREGISTER_CONFIG'

export interface AppWrapConfig {
  readonly withoutStatusBar: boolean
  readonly loading: boolean
  readonly dark: boolean
  readonly locked: boolean
}
const initialAppWrapAttrs: AppWrapConfig = {
  loading: false,
  locked: true,
  withoutStatusBar: false,
  dark: false,
}

export const isAppWrapConfigAttr = (key: string) => initialAppWrapAttrs[key] !== undefined
export const pickAppWrapConfigAttrs: (o: any) => AppWrapConfig =
  pickBy((v, k) => isAppWrapConfigAttr(k))

export interface AppWrapState {
  readonly appWrapConfig: AppWrapConfig
  readonly appWrapConfigsSet: AppWrapConfig[]
}
const initialAppWrapState: AppWrapState = {
  appWrapConfig: initialAppWrapAttrs,
  appWrapConfigsSet: []
}

// TODO(mnzaki): this code is unnecessarily complicated
// the complicated "ConfigsSet" keeping is to support nested <Wrapper>
// components
// - but we don't really need to support them
// - see 20APR[RN61.5] on notion.so:spaces/mnzaki

export const appWrapReducer = (
  state = initialAppWrapState,
  action: AnyAction,
): AppWrapState => {
  let appWrapConfig, appWrapConfigsSet
  switch (action.type) {
    case APPWRAP_REGISTER_CONFIG:
      const idx = state.appWrapConfigsSet.indexOf(action.value)
      if (idx > -1 || Object.keys(action.value).length === 0) return state
      appWrapConfig = {...state.appWrapConfig, ...action.value }
      appWrapConfigsSet = [...state.appWrapConfigsSet, action.value]
      return { appWrapConfig, appWrapConfigsSet }
    case APPWRAP_UNREGISTER_CONFIG:
      const idx2 = state.appWrapConfigsSet.indexOf(action.value)
      if (idx2 < 0) return state
      appWrapConfigsSet = state.appWrapConfigsSet.filter(s => s !== action.value)
      appWrapConfig = appWrapConfigsSet.reduce((prev, cur) => ({...prev, ...cur}), initialAppWrapAttrs)
      return { appWrapConfig, appWrapConfigsSet }

    case APPWRAP_SHOW_LOADER:
      return {...state, appWrapConfig: {...state.appWrapConfig, loading: action.value} }
    case APPWRAP_SET_LOCKED:
      return {...state, appWrapConfig: {...state.appWrapConfig, locked: action.value} }
    default:
      return state
  }
}
