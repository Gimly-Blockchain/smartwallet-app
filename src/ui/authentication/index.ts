export {
  AuthenticationConsent,
} from 'src/ui/authentication/container/authenticationConsent'
