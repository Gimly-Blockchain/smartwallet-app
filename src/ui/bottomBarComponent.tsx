import React from 'react'
import { StyleSheet, View, Dimensions, Platform, TouchableOpacity } from 'react-native'
import Svg, {
  Circle,
  Defs,
  G,
  Mask,
  Path,
  Polygon,
  Use,
} from 'react-native-svg'
import { Colors } from '../styles'
import LinearGradient from 'react-native-linear-gradient'
import { BottomTabBarProps } from 'react-navigation'
import { debug } from '../styles/presets'

const styles = StyleSheet.create({
  wrapper: {
    position: 'absolute',
    width: '100%',
    bottom: -50,
  },
  circle: {
    backgroundColor: Colors.joloColor,
    borderRadius: 35,
    position: 'absolute',
  },
  buttonWrapper: {
    position: 'absolute',
    width: '100%',
    top: 0,
    height: 60,
    zIndex: 2,
    flexDirection: 'row',
    ...debug,
  },
})

// ramda?
const parsePath = () => {
  const origPath =
    'M414,38.2468806 L414,109.427911 L414,109.427911 L0,110.544855 L0,38.2468806 C-2.53876144e-15,17.5163202 16.8054396,0.710880602 37.536,0.710880602 L163.503362,0.710880602 L163.503362,0.710880602 C161.782566,5.63732363 160.846285,10.9383896 160.846285,16.4607145 C160.846285,42.5558923 181.753319,63.710216 207.543594,63.710216 C233.333869,63.710216 254.240903,42.5558923 254.240903,16.4607145 C254.240903,10.9383896 253.304622,5.63732363 251.583826,0.710880602 L376.423017,0.710880602 C397.153577,0.710880602 414,17.5163202 414,38.2468806 Z'
  const origWidth = 414
  const { width } = Dimensions.get('window')

  const coordArray = origPath.split(' ')
  coordArray.pop()
  const coordinates = coordArray.map(vals => {
    const coordinates = vals.split(',')
    const paths = coordinates.map(val => {
      let newVal = val
      let pathType
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      if (isNaN(val[0] as any)) {
        pathType = val[0]
        newVal = val.substr(1)
      } else {
        pathType = ''
      }
      const numVal = parseFloat(newVal)
      const scaledVal = numVal * (width / origWidth)
      return `${pathType}${scaledVal}`
    })
    return paths.join(',')
  })
  coordinates.push('Z')
  return coordinates.join(' ')
}

const SVGBar = () => {
  const { width } = Dimensions.get('window')

  return (
    <Svg width={width} height={110} viewBox={`0 0 ${width} 110`}>
      <Defs>
        <Path
          //d="M414,38.2468806 L414,109.427911 L414,109.427911 L0,110.544855 L0,38.2468806 C-2.53876144e-15,17.5163202 16.8054396,0.710880602 37.536,0.710880602 L163.503362,0.710880602 L163.503362,0.710880602 C161.782566,5.63732363 160.846285,10.9383896 160.846285,16.4607145 C160.846285,42.5558923 181.753319,63.710216 207.543594,63.710216 C233.333869,63.710216 254.240903,42.5558923 254.240903,16.4607145 C254.240903,10.9383896 253.304622,5.63732363 251.583826,0.710880602 L376.423017,0.710880602 C397.153577,0.710880602 414,17.5163202 414,38.2468806 Z"
          d={parsePath()}
          id="path-1"
        />
      </Defs>
      <G id="Symbols" fill="none" fillRule="evenodd">
        <G id="balck_bckg">
          <Mask id="mask-2" fill="white">
            <Use xlinkHref="#path-1" />
          </Mask>
          <G id="Clip-86" />
          <Polygon
            id="Fill-85"
            fill="#0B030D"
            mask="url(#mask-2)"
            points="0 -5.77549882 416.208 -5.77549882 416.208 109.272 6.16191739e-14 109.272"
          />
        </G>
      </G>
    </Svg>
  )
}

interface Props extends BottomTabBarProps {}

export const BottomBarComponent = (props: Props) => {
  console.log(props)
  const { width } = Dimensions.get('window')
  const buttonMarginModif = 16 / 414
  const buttonSizeModif = 0.175
  const circleSize = width * buttonSizeModif

  const buttonVerticalPosition = -Math.floor(
    circleSize / 2 - width * buttonMarginModif,
  )
  const buttonHorizPosition = Math.ceil((width - circleSize) / 2)

  return (
    <View style={styles.wrapper}>
      <View style={styles.buttonWrapper}>
        <View
          style={{ flex: 1, marginLeft: '6%', marginRight: '12%', ...debug }}
        />
        <View
          style={{ flex: 1, marginLeft: '12%', marginRight: '6%', ...debug }}
        />
      </View>
      <TouchableOpacity
        activeOpacity={0.4}
        style={{
          ...styles.circle,
          width: circleSize,
          height: circleSize,
          left: buttonHorizPosition,
          top: buttonVerticalPosition,
        }}
      >
        <LinearGradient
          style={{ flex: 1, borderRadius: 35 }}
          start={{ x: 0, y: 0 }}
          end={{ x: 1, y: 0 }}
          colors={['rgb(145, 25, 66)', 'rgb(210, 45, 105)']}
        />
      </TouchableOpacity>
      <SVGBar />
    </View>
  )
}
