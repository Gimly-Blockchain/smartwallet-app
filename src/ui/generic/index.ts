export { Exception } from './exception'
export { LoadingScreen } from './loading'
export { LoadingSpinner } from './loadingSpinner'
export { SlidingTabBar } from './slidingTabBar'
