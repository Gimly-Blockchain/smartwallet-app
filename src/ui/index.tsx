export { RegistrationProgress } from './registration/containers/loading'
