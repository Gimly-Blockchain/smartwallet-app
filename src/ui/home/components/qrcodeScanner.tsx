import React from 'react'
import { Text, StyleSheet, Platform, BackHandler } from 'react-native'
import { Container } from 'src/ui/structure'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import { Button } from 'react-native-material-ui'
import { QrScanEvent } from '../containers/types'
import I18n from 'src/locales/i18n';
const QRScanner = require('react-native-qrcode-scanner').default

interface Props {
  onScannerSuccess: (e: QrScanEvent) => void
  onScannerCancel: () => void
}

interface State { }

const styles = StyleSheet.create({
  buttonText: {
    color: JolocomTheme.primaryColorBlack
  }
})

export class QRcodeScanner extends React.Component<Props, State> {
  componentDidMount() {
    if (Platform.OS === 'android') {
      BackHandler.addEventListener('hardwareBackPress', () => {
        this.props.onScannerCancel()
      })
    }
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.props.onScannerCancel)
  }

  render() {
    const { onScannerSuccess, onScannerCancel } = this.props
    return (
      <Container>
        <QRScanner
          onRead={(e: QrScanEvent) => onScannerSuccess(e)}
          topContent={<Text>{ I18n.t('You can scan the qr code now!') }</Text>}
          bottomContent={(
            <Button
              onPress={() => onScannerCancel()}
              style={{ text: styles.buttonText }}
              text={ I18n.t("Cancel")}
            />
          )}
        />
      </Container>
    )
  }
}
