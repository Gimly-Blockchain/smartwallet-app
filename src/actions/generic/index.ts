import { Linking } from 'react-native'
import SplashScreen from 'react-native-splash-screen'
import I18n from 'src/locales/i18n'
import { routeList } from 'src/routeList'
import { navigationActions, accountActions } from 'src/actions'
import { ThunkAction } from 'src/store'
import settingKeys from 'src/ui/settings/settingKeys'
import { withLoading, withErrorScreen } from '../modifiers'
import {
  AppWrapConfig,
  APPWRAP_UPDATE_CONFIG,
  APPWRAP_SHOW_LOADER,
  APPWRAP_REGISTER_CONFIG,
  APPWRAP_UNREGISTER_CONFIG,
} from 'src/reducers/generic'
import { AnyAction } from 'redux'

// Default delay on the loading state value before it can switch back to 'false'
const DEFAULT_LOADING_LATCH_DELAY_MS = 300
import { AppError, ErrorCode } from '@jolocom/sdk/js/src/lib/errors'
import { AsyncStorage } from 'react-native'
import crypto from 'crypto'
import { termsOfServiceDE } from 'src/ui/termsAndPrivacy/legalTexts'

export const showErrorScreen = (
  error: AppError | Error,
): ThunkAction => dispatch => {
  const appError: AppError =
    error.constructor === AppError
      ? (error as AppError)
      : new AppError(ErrorCode.Unknown, error)

  return dispatch(
    navigationActions.navigate({
      routeName: routeList.Exception,
      params: {
        returnTo: appError.navigateTo,
        error: appError,
      },
    }),
  )
}

export const initApp: ThunkAction = async (
  dispatch,
  getState,
  backendMiddleware,
) => {
  try {
    const storedSettings = await backendMiddleware.storageLib.get.settingsObject()

    // locale setup
    /** locale setup
     * @dev Until German and Dutch terms are polished, only English is used.
     * previous code:
     */
    if (storedSettings.locale) I18n.locale = storedSettings.locale
    else storedSettings.locale = I18n.locale

    storedSettings.locale = I18n.locale

    await dispatch(loadSettings(storedSettings))

    const ret = await dispatch(accountActions.checkIdentityExists)

    // FIXME what happens if no identity and this is a deeplink?
    // navigationActions.handleDeepLink throws NoWallet
    // need to improve this UX

    // FIXME: get rid of these after setting up deepLinking properly using
    // react-navigation
    const handleDeepLink = (url: string) =>
      dispatch(
        withLoading(withErrorScreen(navigationActions.handleDeepLink(url))),
      )
    Linking.addEventListener('url', event => handleDeepLink(event.url))
    await Linking.getInitialURL().then(url => {
      if (url) handleDeepLink(url)
    })

    return ret
  } catch (e) {
    return dispatch(
      showErrorScreen(
        new AppError(ErrorCode.WalletInitFailed, e, routeList.Landing),
      ),
    )
  } finally {
    SplashScreen.hide()
  }
}

const hashString = (text: string) => {
  return crypto
    .createHash('sha256')
    .update(text)
    .digest('hex')
}

const TERMS_OF_CONDITIONS_KEY = 'TERMS_OF_CONDITIONS'
export const checkTermsOfService = (
  route: routeList,
  onSubmit?: () => void,
): ThunkAction => async dispatch => {
  const storageHash = await AsyncStorage.getItem(TERMS_OF_CONDITIONS_KEY)
  const currentHash = hashString(termsOfServiceDE)
  const shouldShowTerms = storageHash !== currentHash

  if (!shouldShowTerms && onSubmit) onSubmit()
  return dispatch(
    navigationActions.navigate({
      routeName: shouldShowTerms ? routeList.TermsOfServiceConsent : route,
      params: { nextRoute: route, onSubmit },
    }),
  )
}

export const storeTermsOfService = (
  route: routeList,
): ThunkAction => async dispatch => {
  const termsHash = hashString(termsOfServiceDE)
  await AsyncStorage.setItem(TERMS_OF_CONDITIONS_KEY, termsHash)

  dispatch(navigationActions.navigate({ routeName: route }))
}

export const loadSettings = (settings: { [key: string]: any }) => ({
  type: 'LOAD_SETTINGS',
  value: settings,
})

export const setLocale = (locale: string): ThunkAction => async (
  dispatch,
  getState,
  backendMiddleware,
) => {
  await backendMiddleware.storageLib.store.setting(settingKeys.locale, locale)
  I18n.locale = locale
  dispatch({ type: 'SET_LOCALE', value: locale })

  // we need to reset the navigator so that all screens are re-rendered with the
  // new locale
  return dispatch(navigationActions.navigatorReset())
}

const setLoading = (value: boolean) => ({
  type: APPWRAP_SHOW_LOADER,
  value,
})

export const updateAppWrapConfig = (value: AppWrapConfig) => ({
  type: APPWRAP_UPDATE_CONFIG,
  value,
})
export const registerAppWrapConfig = (value: AppWrapConfig & {}) => ({
  type: APPWRAP_REGISTER_CONFIG,
  value,
})
export const unregisterAppWrapConfig = (value: AppWrapConfig) => ({
  type: APPWRAP_UNREGISTER_CONFIG,
  value,
})

export const showAppLoading = (
  shouldShow: boolean,
  latchDelay = DEFAULT_LOADING_LATCH_DELAY_MS,
): ThunkAction => async (dispatch, getState) => {
  const wasShowing = getState().generic.appWrapConfig.loading
  if (shouldShow && !wasShowing) {
    // for setting to "true" we do it immediately and set current state
    const ret = dispatch(setLoading(shouldShow))
    return new Promise<AnyAction>(resolve => {
      setTimeout(() => resolve(ret), latchDelay)
    })
  }

  if (wasShowing && !shouldShow) {
    // for setting to 'false', we delay by the latchDelay amount
    return dispatch(delayedSetLoading(shouldShow, latchDelay))
  }
}

let nextLoadingState = false
let timeout: number | undefined
const delayedSetLoading = (
  value: boolean,
  latchDelay: number,
): ThunkAction => async dispatch => {
  nextLoadingState = value
  if (timeout !== undefined) clearTimeout(timeout)
  timeout = setTimeout(() => {
    dispatch(setLoading(nextLoadingState))
    timeout = undefined
  }, latchDelay)
}
