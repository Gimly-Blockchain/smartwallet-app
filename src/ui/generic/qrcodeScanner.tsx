import React from 'react'
import { StyleSheet, Text } from 'react-native'
import { connect } from 'react-redux'
import { Container } from 'src/ui/structure'
import { Button } from 'react-native-material-ui'
import { QrScanEvent } from 'src/ui/generic/qrcodeScanner'
import I18n from 'src/locales/i18n'
import strings from 'src/locales/strings'
import { JolocomLib } from 'jolocom-lib'
import { interactionHandlers } from 'src/lib/storage/interactionTokens'
import { ThunkDispatch } from 'src/store'
import { showErrorScreen } from 'src/actions/generic'
import { withLoading, withErrorScreen } from 'src/actions/modifiers'
import { NavigationScreenProps } from 'react-navigation'
import { AppError, ErrorCode } from 'src/lib/errors'
import { Colors } from 'src/styles'
import { handelReceiveShard } from '../../actions/recovery'

const QRScanner = require('react-native-qrcode-scanner').default

export interface QrScanEvent {
  data: string
}

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    NavigationScreenProps {}

interface State {}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.backgroundLightMain,
  },
  buttonText: {
    color: Colors.blackMain,
  },
})

export class QRcodeScanner extends React.Component<Props, State> {
  private scanner: typeof QRScanner
  private removeFocusListener: (() => void) | undefined

  constructor(props: Props) {
    super(props)
    if (this.props.navigation) {
      this.removeFocusListener = this.props.navigation.addListener(
        'willFocus',
        () => {
          this.scanner.reactivate()
          // NOTE: force an update to force remounting of the Camera
          this.forceUpdate()
        },
      ).remove
    }
  }

  componentWillUnmount() {
    if (this.removeFocusListener) this.removeFocusListener()
  }

  onScannerCancel() {
    if (this.props.navigation) this.props.navigation.goBack()
  }

  render() {
    const { onScannerSuccess } = this.props
    // NOTE: the key is used to invalidate the previously rendered component as
    // we need to rerender and remount the camera to ensure it is properly setup
    const cameraProps = { key: Date.now() }
    return (
      <React.Fragment>
        <Container style={styles.container}>
          <QRScanner
            cameraProps={cameraProps}
            ref={(ref: React.Component) => (this.scanner = ref)}
            onRead={onScannerSuccess}
            cameraStyle={{ overflow: 'hidden' }}
            topContent={
              <Text>{I18n.t(strings.YOU_CAN_SCAN_THE_QR_CODE_NOW)}</Text>
            }
            bottomContent={
              <Button
                onPress={this.onScannerCancel.bind(this)}
                style={{ text: styles.buttonText }}
                text={I18n.t(strings.CANCEL)}
              />
            }
          />
        </Container>
      </React.Fragment>
    )
  }
}

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  onScannerSuccess: async (e: QrScanEvent) => {
    let interactionToken
    const shardPrefix = 'shard:'
    if (e.data.startsWith(shardPrefix)) {
      interactionToken = e.data.slice(shardPrefix.length)
      return dispatch(handelReceiveShard(interactionToken))
    }
    try {
      interactionToken = JolocomLib.parse.interactionToken.fromJWT(e.data)
    } catch (err) {
      return dispatch(
        showErrorScreen(new AppError(ErrorCode.ParseJWTFailed, err)),
      )
    }

    const handler = interactionHandlers[interactionToken.interactionType]

    return handler
      ? dispatch(withLoading(withErrorScreen(handler(interactionToken))))
      : dispatch(
          showErrorScreen(
            new AppError(ErrorCode.Unknown, new Error('No handler found')),
          ),
        )
  },
})

export const QRScannerContainer = connect(
  null,
  mapDispatchToProps,
)(QRcodeScanner)
