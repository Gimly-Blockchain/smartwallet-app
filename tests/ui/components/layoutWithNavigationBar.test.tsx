import React from 'react'
import { shallow } from 'enzyme'
import { Text } from 'react-native'

import { LayoutWithNavigationBar } from 'src/ui/generic'
import { QRcodeScanner } from 'src/ui/generic/qrcodeScanner'
import { LoadingSpinner } from 'src/ui/generic'


describe('LayoutWithNavigationBar component', () => {
  const COMMON_PROPS = {
    onScannerSuccess: (jwt: string) => null,
    loading: false,
  }

  it('should display only qr code scanner based on state', () => {
    const layoutComponent = shallow((
      <LayoutWithNavigationBar {...COMMON_PROPS}>
        <Text>Sample test children</Text>
      </LayoutWithNavigationBar>
    ))
    layoutComponent.setState({ scanning: true })

    expect(layoutComponent.find(QRcodeScanner).length).toBe(1)
    expect(layoutComponent.find(Text).length).toBe(0)
    expect(layoutComponent).toMatchSnapshot()
  })

  it('should display only loading spinner based on state', () => {
    const layoutComponent = shallow((
      <LayoutWithNavigationBar {...COMMON_PROPS}>
        <Text>Sample test children</Text>
      </LayoutWithNavigationBar>
    ))
    layoutComponent.setState({ loading: true })

    expect(layoutComponent.find(LoadingSpinner).length).toBe(1)
    expect(layoutComponent.find(Text).length).toBe(0)
    expect(layoutComponent).toMatchSnapshot()
  })

  it('should display only loading spinner based on props', () => {
    const props = Object.assign({}, COMMON_PROPS, {
      loading: true
    })
    const layoutComponent = shallow((
      <LayoutWithNavigationBar {...props}>
        <Text>Sample test children</Text>
      </LayoutWithNavigationBar>
    ))

    expect(layoutComponent.find(LoadingSpinner).length).toBe(1)
    expect(layoutComponent.find(Text).length).toBe(0)
    expect(layoutComponent).toMatchSnapshot()
  })

  it('should render children with qr code button in a navigation if no loading and scanning state', () => {
    const layoutComponent = shallow((
      <LayoutWithNavigationBar>
        <Text>Sample test children</Text>
      </LayoutWithNavigationBar>
    ))

    expect(layoutComponent.find('Text').children().text()).toBe('Sample test children')
    expect(layoutComponent).toMatchSnapshot()
  })
})
