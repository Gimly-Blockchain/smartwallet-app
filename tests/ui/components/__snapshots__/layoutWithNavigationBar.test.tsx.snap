// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutWithNavigationBar component should display only loading spinner based on props 1`] = `<LoadingSpinner />`;

exports[`LayoutWithNavigationBar component should display only loading spinner based on state 1`] = `<LoadingSpinner />`;

exports[`LayoutWithNavigationBar component should display only qr code scanner based on state 1`] = `
<QRcodeScanner
  onScannerCancel={[Function]}
  onScannerSuccess={[Function]}
/>
`;

exports[`LayoutWithNavigationBar component should render children with qr code button in a navigation if no loading and scanning state 1`] = `
<Component>
  <Component
    style={
      Object {
        "paddingBottom": 55,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Sample test children
    </Text>
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "bottom": 0,
        "height": 82,
        "justifyContent": "flex-end",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafafa",
          "flexDirection": "row",
          "height": 55,
          "justifyContent": "space-around",
          "width": "100%",
        }
      }
    >
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-around",
            },
            Object {
              "marginRight": 36,
            },
          ]
        }
      />
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-around",
            },
            Object {
              "marginLeft": 36,
            },
          ]
        }
      />
    </Component>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#942f51",
          "borderRadius": 35,
          "bottom": 6,
          "elevation": 8,
          "height": 72,
          "justifyContent": "center",
          "position": "absolute",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 72,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="white"
        name="qrcode-scan"
        size={30}
      />
    </TouchableOpacity>
  </Component>
</Component>
`;
